tests/ballerina-test/src/test/resources/test-src/types/stream/stream-value.bal

```diff
@@ -20,21 +22,28 @@ function testInvalidObjectPublishingToStream () {
     s1.publish(j1);
 }
 
-function testSubscriptionFunctionWithNonObjectParameter () {
-    stream<Employee> s1;
-    s1.subscribe(printInteger);
-}
-
 function testSubscriptionFunctionWithIncorrectObjectParameter () {
     stream<Employee> s1;
     s1.subscribe(printJobDescription);
 }
 
+int arrayIndex = 0;
 Employee globalEmployee;
-Employee[] globalEmployeeArray = [];
-int employeeIndex = 0;
 
-function testStreamPublishingAndSubscription () returns (Employee, Employee, Employee) {
+function testGlobalStream () returns (Employee, Employee, Employee) {
+    Employee origEmployee = globalEmployee;
+    globalEmployeeStream.subscribe(assignGlobalEmployee);
+    Employee publishedEmployee = { id:5678, name:"Maryam" };
+    globalEmployeeStream.publish(publishedEmployee);
+    int startTime = time:currentTime().time;
+    while (globalEmployee.id == 0 && time:currentTime().time - startTime < 1000) {
+        //allow for value update
+    }
+    Employee newEmployee = globalEmployee;
+    return (origEmployee, publishedEmployee, newEmployee);
```

Comment:
Can't we return `globalEmployee` directly here? Why do we need to create a new `Employee`?

Reply:
Fixed in a02c67b